### PR TITLE
Webhook response

### PR DESCRIPTION
**Pull request by hotmeteor**
Returns `\GuzzleHttp\Psr7\Response` from a successful webhook request rather than parsing the status code and succeeding silently or throwing an exception.

This functionality allows the app to handle response data itself and creating logging and further action based on statuses or data.

### CODE REVIEW

File: src/WebhookChannel.php [hunk 1]
```diff
@@ -45,5 +47,7 @@ public function send($notifiable, Notification $notification)
         if ($response->getStatusCode() >= 300 || $response->getStatusCode() < 200) {
             throw CouldNotSendNotification::serviceRespondedWithAnError($response);
```
Comment (atymic) [inline src/WebhookChannel.php:49]: I still like the idea of setting the response on the exception. Are you able to do that?

Comment (hotmeteor) [reply src/WebhookChannel.php:49]: Ya, but not sure I can do it and maintain that `serviceRespondedWithAnError` method. Seems like I'd need to construct it differently and add a `getResponse` method. But maybe I'm wrong?

Comment (atymic) [reply src/WebhookChannel.php:49]: Something like this

```php
use GuzzleHttp\Psr7\Response;

class CouldNotSendNotification extends \Exception
{
    /** @var Response */
    private $response;

    public function __construct(Response $response, string $message, int $code = null)
    {
        $this->response = $response;
        $this->message = $message;
        $this->code = $code ?? $response->getStatusCode();
    }


    public static function serviceRespondedWithAnError(Response $response)
    {
        return new self(
            $response,
            sprintf('Webhook responded with an error: `%s`', $response->getBody()->getContents())
        );
    }
}
```

Comment (hotmeteor) [reply src/WebhookChannel.php:49]: Done.

File: src/Exceptions/CouldNotSendNotification.php [hunk 2]
```diff
@@ -2,10 +2,44 @@
...
 
+use GuzzleHttp\Psr7\Response;
+
 class CouldNotSendNotification extends \Exception
 {
-    public static function serviceRespondedWithAnError($response)
+    private $response;
+
+    /**
+     * CouldNotSendNotification constructor.
```
Comment (atymic) [inline src/Exceptions/CouldNotSendNotification.php]: Remove comment please :)

File: src/Exceptions/CouldNotSendNotification.php [hunk 3]
```diff
@@ -2,10 +2,44 @@
...
+        $this->response = $response;
+        $this->message = $message;
+        $this->code = $code ?? $response->getStatusCode();
+
+        parent::__construct($message, $code);
+    }
+
+    /**
+     * @param Response $response
+     * @return CouldNotSendNotification
```
Comment (atymic) [inline src/Exceptions/CouldNotSendNotification.php]:
```suggestion
     * @return self
```

**Review by atymic — COMMENTED** [review]
Can we add a test to ensure that the response is on the exception/has the correct message & code please?

**Comment by hotmeteor** [comment]
How's that look?

**Comment by atymic** [comment]
Looks good, but looks like travis is failing due to an issue with the test.

**Comment by hotmeteor** [comment]
My bad

**Comment by atymic** [comment]
And styleci 😅

**Comment by hotmeteor** [comment]
😪 There we go

**Comment by atymic** [comment]
Thanks for PR :)
Have merged it, but I can't tag it as packagist is broken :( #29

**Pull request merged**